os/board/common/crashdump.c

```diff
@@ -58,6 +82,176 @@
  * Private Functions
  ****************************************************************************/
 
+#if defined(CONFIG_BOARD_RAMDUMP_FLASH) || defined(CONFIG_BOARD_COREDUMP_FLASH)
+static int get_dump_partname(char *part_name)
+{
+	FAR struct mtd_dev_s *mtd;
+	static char temp[MTD_PARTNAME_LEN + 1];
+	int index = 0;
+
+	do {
+		mtd = get_mtd_partition(index);
+		if (mtd) {
+			mtd_getpartitionname(mtd, temp);
+			if (!strcmp(temp, part_name))
+				return index;
+		}
+		index++;
+	} while (mtd);
+
+	return -1;
+}
+#endif
+
+#if defined(CONFIG_BOARD_COREDUMP_FLASH)
+int coredump_to_flash(uint32_t cur_sp, void *tcb)
```

Comment:
You didn't check the return value of this function.
How about changing the return type or checking the return when you use?